android/KMEA/app/src/main/java/com/keyman/engine/KMKeyboardJSHandler.java

```diff
@@ -284,7 +285,17 @@ private static void performLeftDeletions(InputConnection ic, int dn) {
 
     // Chop dn+numPairs code points from the end of charsBackup
     // subSequence indices are start(inclusive) to end(exclusive)
-    CharSequence expectedChars = charsBackup.subSequence(0, charsBackup.length() - (dn + numPairs));
+    int start = 0;
+    int end = charsBackup.length() - (dn + numPairs);
+    CharSequence expectedChars;
+    try {
+      expectedChars = charsBackup.subSequence(start, end);
+    } catch (IndexOutOfBoundsException e) {
+      KMLog.LogException(TAG,
+        KMString.format("Bad subSequence of start %d, end is %d, length %d, dn %d, numPairs %d",
+        start, end, charsBackup.length(), dn, numPairs), e);
+      expectedChars = charsBackup;
```

Comment:
```suggestion
      expectedChars = "";
```

We can't delete _past_ the boundary, but we can certainly delete _up to_ the boundary.  Once #10869 is in place, that would probably be the best way to handle unexpected behavior here.